### PR TITLE
[LD-100] Authomatic snapshots tests from compose previews. [New snapshots]

### PR DESCRIPTION
**Pull request by Krzysiudan**
## Why? 

Snapshots can be created from every compose preview in a module. That's a lot of time-saving if there wouldn't be a need for creating additional paparazzi tests. 

## Changes
- Added Showkase library 
- Added configuration for automatic tests from the compose previews
- Added testParameterInjector dependency, which is the main clue for making the tests work 
- Removed previous tests `LoudiusButtonTest` and `LoudiusDialogTests` since they are tested now anyway. 

## How 
Test parameter injector allows to creation multiple unit tests from a single @Test method. Thanks to that we can have multiple tests from all the @Preview composable.  There is an additional bonus - you can have screenshots for dark & light modes, different font sizes or locales very easily. 

## Note
1. You can disable particular previews from testing by doing it `private`. Please comment on screenshots you find not valuable to have and to test.  
2. That's not my idea, all of the credits should go to @davidvavra and his article : 
https://proandroiddev.com/automatic-screenshot-testing-for-all-your-compose-previews-6add202fecc7

### CODE REVIEW

File: components/build.gradle [hunk 1]
```diff
@@ -28,6 +28,12 @@ android {
         kotlinCompilerExtensionVersion '1.4.2'
     }
     resourcePrefix "components_"
+
+    kapt {
+        arguments {
+            arg("skipPrivatePreviews", "true")
+        }
+    }
```
Comment (Krzysiudan) [inline components/build.gradle:36]: **Info** To make skipping private previews possible.

File: components/src/test/java/com/appunite/loudius/ShowkaseRoot.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * limitations under the License.
+ */
+
+package com.appunite.loudius
+
+import com.airbnb.android.showkase.annotation.ShowkaseRoot
+import com.airbnb.android.showkase.annotation.ShowkaseRootModule
+
+@ShowkaseRoot
+class ShowkaseRoot : ShowkaseRootModule
```
Comment (nowakweronika) [inline components/src/test/java/com/appunite/loudius/ShowkaseRoot.kt:23]: Question: Where do you use it?

Comment (Krzysiudan) [reply components/src/test/java/com/appunite/loudius/ShowkaseRoot.kt:23]: `Showkase.getMedaData()` in a Paparazzi test

**Review by kezc — APPROVED** [review]

File: components/src/test/snapshots/images/com.appunite.loudius_PaparazziShowkaseTests_preview_tests[Default Group:PreviewLoudiusPlaceholderText,2,light].png
Comment (marcin-adamczewski) [inline components/src/test/snapshots/images/com.appunite.loudius_PaparazziShowkaseTests_preview_tests[Default Group:PreviewLoudiusPlaceholderText,2,light].png:1]: There is a new line character visible in text -> \n. I guess it's an issue?

Comment (Krzysiudan) [reply components/src/test/snapshots/images/com.appunite.loudius_PaparazziShowkaseTests_preview_tests[Default Group:PreviewLoudiusPlaceholderText,2,light].png:1]: Nicely spotted. That's a typo in the @Preview string only. I've done the fix in a separate PR, not to make this PR clumsy.  
https://github.com/appunite/Loudius/pull/105

**Review by nowakweronika — APPROVED** [review]
lgtm :)

**Pull request merged**